### PR TITLE
Solved Task003-005; all classes - upgraded

### PR DESCRIPTION
**Pull request by kmavropulo**
:camel: In this request
Solved:
- Task003-005;

Created:
- Task003Impl-005Imp;
- Validator

Upgraded:
- Task004Impl;
- Task003Impl;
- Task002Impl;
- Task001Impl;
- MappingOperationImpl;
- Validator

### CODE REVIEW

File: src/main/java/com/epam/university/java/core/task001/Task001Impl.java [hunk 1]
```diff
@@ -1,134 +1,64 @@
...
 
+import com.epam.university.java.core.utils.Validator;
+
 public class Task001Impl implements Task001 {
-    public static final String firstNumberMessageForIllegalArgumentException = "first number must be the value";
-    public static final String secondNumberMessageForIllegalArgumentException = "second number must be the value";
-    public static final String firstNumberMessageForNumberFormatException = "can't convert first input value to number";
-    public static final String secondNumberMessageForNumberFormatException = "can't convert second input value to number";
+
+    public static final String errorMessageForFirstParameterIfLengthNull =
```
Comment (aabarmin) [inline src/main/java/com/epam/university/java/core/task001/Task001Impl.java]: Names of constants should be in upper case, ex. ERROR_MESSAGE_FOR_NULL

File: src/main/java/com/epam/university/java/core/task003/FilteringConditionImpl.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.epam.university.java.core.task003;
+
+public class FilteringConditionImpl implements FilteringCondition {
+    @Override
+    public boolean isValid(String element) {
+        if(element=="One"|element=="Two"|element==null){
```
Comment (aabarmin) [inline src/main/java/com/epam/university/java/core/task003/FilteringConditionImpl.java]: It's better to compare literals with equals, ex. "One".equals(element). And it is preferably to use short form of disjunctive operation.

File: src/main/java/com/epam/university/java/core/task001/Task001Impl.java [hunk 3]
```diff
@@ -1,134 +1,64 @@
...
-        } catch (NumberFormatException ex) {
-            throw new NumberFormatException(firstNumberMessageForNumberFormatException);
-        }
-        try {
-            secondNumberConverted = Double.parseDouble(secondNumber);
-        } catch (NumberFormatException ex) {
-            throw new NumberFormatException(secondNumberMessageForNumberFormatException);
-        }
-        return firstNumberConverted + secondNumberConverted;
+        Validator.validateValueAndLengthNotNull(firstNumber, secondNumber,
```
Comment (aabarmin) [inline src/main/java/com/epam/university/java/core/task001/Task001Impl.java:9]: It's preferably to store default values of error messages in validator itself.

Comment (kmavropulo) [reply src/main/java/com/epam/university/java/core/task001/Task001Impl.java:9]: Respected Alexander Barmin!
Thank u for the comments - all fulfiled:
- Task005 - complited; 
- Validator and all it's users have been enhanced 
(new default messages added;
 literals comparing by equals; 
using forms of disjunctive operators where possible.)

**Pull request merged**